operators/src/engine/operator.rs

```diff
@@ -65,21 +43,236 @@ pub trait VectorOperator: Operator + CloneableVectorOperator {
 /// Common methods for `RasterOperator`s
 #[typetag::serde(tag = "type")]
 pub trait RasterOperator: Operator + CloneableRasterOperator {
+    fn initialized_operator(
+        self: Box<Self>,
+        context: ExecutionContext,
+    ) -> Result<Box<dyn InitializedRasterOperator>>;
+
+    /// Wrap a box around a `RasterOperator`
+    fn boxed(self) -> Box<dyn RasterOperator>
+    where
+        Self: Sized + 'static,
+    {
+        Box::new(self)
+    }
+}
+
+pub type ExecutionContext = u32;
```

Comment:
Should be an empty struct I guess. Isn't the context something rather for the query method?

Reply:
this context is required for information like data location and user information. One use-case is the following: Some of the sources need to check with a database and /or config file. Using this context, we can keep the config management separated from the operators.

Reply:
converted it into a struct.